ext/storage_sources/gcp_store/gcp_log_system.h

```diff
@@ -0,0 +1,90 @@
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include <wiredtiger.h>
+#include <wiredtiger_ext.h>
+
+#include <google/cloud/log.h>
+
+#include <atomic>
+
+// Mapping the desired WiredTiger extension verbosity level to a rough equivalent GCP
+// SDK verbosity level.
+static const std::map<int32_t, google::cloud::Severity> verbosity_mapping = {
+  {WT_VERBOSE_ERROR, google::cloud::Severity::GCP_LS_ERROR},
+  {WT_VERBOSE_WARNING,  google::cloud::Severity::GCP_LS_WARNING},
+  {WT_VERBOSE_INFO, google::cloud::Severity::GCP_LS_INFO},
+  {WT_VERBOSE_DEBUG_1, google::cloud::Severity::GCP_LS_DEBUG},
+  {WT_VERBOSE_DEBUG_2, google::cloud::Severity::GCP_LS_DEBUG},
+  {WT_VERBOSE_DEBUG_3, google::cloud::Severity::GCP_LS_DEBUG},
+  {WT_VERBOSE_DEBUG_4, google::cloud::Severity::GCP_LS_DEBUG},
+  {WT_VERBOSE_DEBUG_5, google::cloud::Severity::GCP_LS_TRACE}
+};
+
+/*
+ * Provides the GCP Store with a logger implementation that redirects the generated logs to
+ * WiredTiger's logging streams. This class implements GCP's LogBackend class, an interface
+ * for logging implementations. Functions are derived from the interface to incorporate the logging
+ * with WiredTiger's logging system. 
+ * 
+ * GCP's LogSink is used to initialize the intialized log system to the SDK.
+ */
+class gcp_log_system : public google::cloud::LogBackend {
+    public:
+    gcp_log_system(WT_EXTENSION_API *wt_api, uint32_t wt_verbosity_level);
+    
+    void Process (google::cloud::LogRecord const &log_record) override;
+    void ProcessWithOwnership (google::cloud::LogRecord log_record) override { Process(log_record); };
+    // Inherited from LogBackend and is not used.
+    void
+    Flush() override final
+    {
+    }
+
+    // Sends error messages to WiredTiger's error level log stream.
+    void
+    log_error_message(const std::string &message) const
+    {
+        log_verbose_message(WT_VERBOSE_ERROR, message);
+    }
+
+    // Sends error messages to WiredTiger's debug level log stream.
+    void
+    log_debug_message(const std::string &message) const
+    {
+        log_verbose_message(WT_VERBOSE_DEBUG_1, message);
+    }
+
+    // Sets the WiredTiger Extension's verbosity level and matches the GCP log levels
+    // to this.
+    void set_wt_verbosity_level(int32_t wtVerbosityLevel);
+    private:
+    void log_verbose_message(int32_t verbosity_level, const std::string &message) const;
+    std::atomic<google::cloud::Severity> _gcp_log_level;
```

Comment:
I don't think this needs to be atomic, right?

Reply:
The reason behind the atomic was grabbed from the GCP docs, https://googleapis.dev/cpp/google-cloud-common/2.7.0/namespacegoogle_1_1cloud.html#aeeb9b9a1eeb3fc7f6ff13bd078172922 describing 
```
We force the enum to be represented as an int because we will store the values in an std::atomic<> and the implementations usually optimize std::atomic<int> but not std::atomic<Foo>
```
My idea was that if the docs do it, its probably best to also store it as an atomic. What are your thoughts @korteland? I am happy to remove as well.

Reply:
I'd prefer to remove it, we don't have any form of multithreading around setting it (to my knowledge?) and the quoted docs are talking about how it's represented inside the library, which we don't need to copy.

Reply:
done! [2936fb9](https://github.com/wiredtiger/wiredtiger/pull/8868/commits/2936fb92a544a27ff55c64e6703520024c193c37)